app/models/rating.rb

```diff
@@ -0,0 +1,46 @@
+class Rating
+  include RedisModel
+  include PresentableModel
+
+  # each CRUD operation updates the current state in Redis but also writes
+  # an event to Kafka
+
+  DATA_FIELDS = [:id, :app_id, :source_domain, :content_uid,
+                 :variant_id, :user_uid, :type, :value,
+                 :created_at]
+
+  attr_reader *DATA_FIELDS
+
+  validates :id, presence: true
+  validates :app_id, presence: true
+  validates :content_uid, presence: true
+  validates :user_uid, presence: true
+  validates :type, presence: true,
+                   inclusion: { in: %w(thumbs five_star full_range) }
+  validates :value, presence: true,
```

Comment:
I"m guessing not, but could :type be an :enum ?  If so that'd ensure that it's only set to the known values

Reply:
I don't think so since this is activemodel, not activerecord?

Reply:
I was thinking maybe we could include them, but it's probably too much effort for just this one case.